File: source/clibCoreCs/addonCollection_core/0/faviconIco.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.Core
    Public Class faviconIcoClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core
                Dim Filename As String = cpCore.siteProperties.getText("FaviconFilename", "")
                If Filename = "" Then
                    '
                    ' no favicon, 404 the call
                    '
                    Call cpCore.webServer.setResponseStatus("404 Not Found")
                    Call cpCore.webServer.setResponseContentType("image/gif")
                    cpCore.doc.continueProcessing = False
                    Return String.Empty
                Else
                    cpCore.doc.continueProcessing = False
                    Return cpCore.webServer.redirect(genericController.getCdnFileLink(cpCore, Filename), "favicon request", False, False)
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: source/clibCoreCs/addonCollection_core/faviconIco.cs
using System;

//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using Contensive.BaseClasses;
using Contensive.Core;
using Models.Entity;
using Contensive.Core.Controllers;

//
namespace Contensive.Addons.Core
{
	public class faviconIcoClass : Contensive.BaseClasses.AddonBaseClass
	{
		//
		//====================================================================================================
		/// <summary>
		/// getFieldEditorPreference remote method
		/// </summary>
		/// <param name="cp"></param>
		/// <returns></returns>
		public override object Execute(Contensive.BaseClasses.CPBaseClass cp)
		{
			string result = "";
			try
			{
				CPClass processor = (CPClass)cp;
				coreClass cpCore = processor.core;
				string Filename = cpCore.siteProperties.getText("FaviconFilename", "");
				if (string.IsNullOrEmpty(Filename))
				{
					//
					// no favicon, 404 the call
					//
					cpCore.webServer.setResponseStatus("404 Not Found");
					cpCore.webServer.setResponseContentType("image/gif");
					cpCore.doc.continueProcessing = false;
					return string.Empty;
				}
				else
				{
					cpCore.doc.continueProcessing = false;
					return cpCore.webServer.redirect(genericController.getCdnFileLink(cpCore, Filename), "favicon request", false, false);
				}
			}
			catch (Exception ex)
			{
				cp.Site.ErrorReport(ex);
			}
			return result;
		}
	}
}
